erroranalysis/erroranalysis/_internal/matrix_filter.py

```diff
@@ -178,21 +179,29 @@ def compute_json_matrix(analyzer, features, filters, composite_filters):
                     fix_counts.append(counts[counts_idx])
                     counts_idx += 1
             counts = fix_counts
-            cutdf_err = pd.cut(df_err[feat1], bins)
-            catr_err = cutdf_err.cat.rename_categories(bin_range)
+            cut_err = pd.cut(df_err[feat1], bins)
+            catr_err = cut_err.cat.rename_categories(bin_range)
             val_err, counts_err = np.unique(catr_err.to_numpy(),
                                             return_counts=True)
-            val_err = cutdf_err.cat.categories[val_err]
-            json_matrix = json_matrix_1d(cutdf.cat.categories,
-                                         val_err, counts,
-                                         counts_err, metric)
+            val_err = cut_err.cat.categories[val_err]
+            categories = cutdf.cat.categories
         else:
-            values, counts = np.unique(df[feat1].to_numpy(),
-                                       return_counts=True)
+            categories, counts = np.unique(df[feat1].to_numpy(),
+                                           return_counts=True)
             val_err, counts_err = np.unique(df_err[feat1].to_numpy(),
                                             return_counts=True)
-            json_matrix = json_matrix_1d(values, val_err, counts,
-                                         counts_err, metric)
+            cut_err = df_err
+        # Compute the given metric for each group, if not using error rate
+        if metric != Metrics.ERROR_RATE:
+            aggfunc = _AggFunc(metric_to_func[metric])
+            cutdf_err = pd.DataFrame(cut_err)
+            cutdf_err['metric_values'] = list(zip(df_err[TRUE_Y],
+                                                  df_err[PRED_Y]))
+            grouped = cutdf_err.groupby([feat1])
+            counts_err = grouped.agg(aggfunc._agg_func_triplet)
+            counts_err = counts_err.values.flatten()
```

Comment:
I'm not super familiar with this code, but what do you think about using `ravel` to use less memory? https://stackoverflow.com/questions/28930465/what-is-the-difference-between-flatten-and-ravel-functions-in-numpy

Reply:
done!